src/external/Collector.ts

```diff
@@ -305,22 +320,30 @@ class Collector extends BaseExternal {
       }
 
       const txQuery = `${this.baseUrl}/api/transaction?blockNumber=${blockNumber}`
-
+      let blockGasUsed = '0x0'
+      // This parameter is used to determine the total gas used of transactions in the block
+      // It is used to calculate the block gas used
       resultBlock.transactions = await axios
         .get(txQuery)
         .then((response) => {
           if (!response.data.success) return []
           return response.data.transactions.map((tx: any) => {
+            let thisDecodeTransaction = this.decodeTransaction(tx)
+            blockGasUsed = BigNumber.from(blockGasUsed).add(thisDecodeTransaction.gas).toHexString()
             //need to review the safety of this for caching and support that this could change!
             // UPDATE: We're now handling the response as per the "details" flag by mutating the "transactions" field. The default cache storage contains the full transaction details.
-            return this.decodeTransaction(tx)
+            return thisDecodeTransaction
           })
         })
         .catch((e) => {
           nestedCountersInstance.countEvent('collector', `getBlock-error tx ${e.message}`)
           console.error('collector.getBlock could not get txs for the block', e)
           return []
         })
+      // Now we have the block gas used then return it to the RPC method
+      resultBlock.gasUsed = blockGasUsed
+      // Start to calculate the transaction root fot this block
+      resultBlock.transactionsRoot = await this.calculateTransactionRoot(readableBlock.tranactions)
```

Comment:
readableBlock.tranactions <-- appears to have a typo here.  Should be "transactions"?